src/coreclr/src/jit/lsraarm64.cpp

```diff
@@ -1011,13 +1011,22 @@ int LinearScan::BuildHWIntrinsic(GenTreeHWIntrinsic* intrinsicTree)
     int srcCount = 0;
     int dstCount = intrinsicTree->IsValue() ? 1 : 0;
 
-    // We may need to allocate an additional general-purpose register when an intrinsic has a non-const immediate
-    // operand and the intrinsic does not have an alternative non-const fallback form.
-    // However, for a case when the operand can take only two possible values - zero and one
-    // the codegen will use cbnz to do conditional branch.
-    bool mayNeedBranchTargetReg = (intrin.category == HW_Category_IMM) && !HWIntrinsicInfo::NoJmpTableImm(intrin.id) &&
-                                  (HWIntrinsicInfo::lookupImmUpperBound(intrin.id, intrinsicTree->gtSIMDSize,
-                                                                        intrinsicTree->gtSIMDBaseType) != 2);
+    bool mayNeedBranchTargetReg = false;
+
+    if ((intrin.category == HW_Category_IMM) && !HWIntrinsicInfo::NoJmpTableImm(intrin.id))
+    {
+        // We may need to allocate an additional general-purpose register when an intrinsic has a non-const immediate
+        // operand and the intrinsic does not have an alternative non-const fallback form.
+        // However, for a case when the operand can take only two possible values - zero and one
+        // the codegen will use cbnz to do conditional branch.
+
+        int immLowerBound = 0;
+        int immUpperBound = 0;
+
+        HWIntrinsicInfo::lookupImmBounds(intrin.id, intrinsicTree->gtSIMDSize, intrinsicTree->gtSIMDBaseType,
+                                         &immLowerBound, &immUpperBound);
+        mayNeedBranchTargetReg = (immLowerBound != 0) || (immUpperBound != 1);
```

Comment:
Could this use `HWIntrinsicImmOpHelper::TestImmOpZeroOrOne`? It belongs to `CodeGen` but might make sense to use it from LSRA as well.

Reply:
I cannot use HWIntrinsicImmOpHelper::TestImmOpZeroOrOne() since it tests the class instance fields immLowerBound and immUpperBound. But I think I can add helper like this to use in both. I will follow up on this